### DISCUSSION TITLE
Using chunk method for every row

### DISCUSSION THREAD

**Discussion by abduqodirovshukrullo** [def-studio/telegraph#174] [Q&A] [answered]
I want button structure like this:
![image](https://user-images.githubusercontent.com/45895327/190370582-f3834d7a-f15e-4f3b-af4f-140cf4ce16c5.png)

I tried this code.

           $specialitieKeyboards = [];
            foreach ($specialities as $specialitiy) {
                $specialitieKeyboards[] = ReplyButton::make($specialitiy->name)->webApp('google.com');
            }
            $this->chat->message('<b>Message</b>')
                ->replyKeyboard(
                    ReplyKeyboard::make()
                    ->row([
                        ReplyButton::make('◀️button'),
                        ReplyButton::make('🔙button')
                    ])->chunk(1)
                    ->row($specialitieKeyboards)->chunk(2)
                        
                    ->row([
                        ReplyButton::make('◀️button'),
                        ReplyButton::make('🔙button')
                    ])->chunk(1)
                )
                ->send();
But chunk() method not working for every row();
can you give a suggestion?

**Comment by fabio-ivona** [comment, marked as answer]
Hi, a better approach could be to use the `width` method of buttons:

```php
$specialitieKeyboards = [];
foreach (specialities as $specialitiy) {
    $specialitieKeyboards[] = ReplyButton::make($specialitiy->name)->webApp('google.com')->width(0.5);
}

Telegraph::message('<b>Message</b>')
    ->replyKeyboard(ReplyKeyboard::make()
        ->button('◀️button')->width(1)
        ->button('🔙button')->width(1)
        ->buttons($specialitieKeyboards)
        ->button('◀️button')->width(1)
        ->button('🔙button')->width(1)
    )->send();
```

**Reply by abduqodirovshukrullo** [reply]
Thank you very much.
